pkg/api/service/finding.go

```diff
@@ -72,12 +65,33 @@ func (s vulcanitoService) ListFindingOverwrites(ctx context.Context, findingID s
 func isValidFindingStatus(status string) bool {
 	// Set of valid status type
 	validStatus := map[string]struct{}{
-		"OPEN":           struct{}{},
-		"FIXED":          struct{}{},
-		"EXPIRED":        struct{}{},
-		"FALSE_POSITIVE": struct{}{},
+		"OPEN":           {},
+		"FIXED":          {},
+		"EXPIRED":        {},
+		"FALSE_POSITIVE": {},
 	}
 
 	_, existsInSet := validStatus[status]
 	return existsInSet
 }
+
+func isValidFindingTransition(status, statusPrevious string) bool {
+	// Valid transitions:
+	//
+	// OPEN           -> OPEN
+	// OPEN           -> FALSE_POSITIVE
+	// FALSE_POSITIVE -> OPEN
+	// FALSE_POSITIVE -> FALSE_POSITIVE
+	// FIXED          -> FALSE_POSITIVE
+	if status == statusPrevious {
+		return true
+	}
+
+	validTransitions := map[string]any{
+		"OPEN->FALSE_POSITIVE":  nil,
+		"FALSE_POSITIVE->OPEN":  nil,
+		"FIXED->FALSE_POSITIVE": nil,
+	}
+	_, valid := validTransitions[fmt.Sprintf("%s->%s", statusPrevious, status)]
+	return valid
```

Comment:
I would do something like this:

```suggestion
	validTransitions := map[string][]string{
		"OPEN":  []string{"FALSE_POSITIVE"},
		"FALSE_POSITIVE":  []string{"OPEN"},
		"FIXED": []string{"FALSE_POSITIVE"},
	}
	return slices.Contains(validTransitions[statusPrevious], status)
```

If we know that there will be always only one valid transition per status, then we can just replace the `map[string][]string` with `map[string]string` and `slices.Contains` with `==`.

Reply:
I like it, thanks